gap/semigroups/semicons.gi

```diff
@@ -805,3 +805,227 @@ for _IsXSemigroup in ["IsTransformationSemigroup",
+                          type,
+                          SemilatticeOfStrongSemilatticeOfSemigroups,
+                          rtclosure,
+                          SemigroupsOfStrongSemilatticeOfSemigroups,
+                          semigroups,
+                          HomomorphismsOfStrongSemilatticeOfSemigroups,
+                          maps,
+                          ElementTypeOfStrongSemilatticeOfSemigroups,
+                          etype,
+                          GeneratorsOfMagma,
+                          gens);
+  return out;
+end);
+
+InstallMethod(Size, "for a strong semilattice of semigroups",
+[IsStrongSemilatticeOfSemigroups],
+function(S)
+  return Sum(SemigroupsOfStrongSemilatticeOfSemigroups(S), Size);
+end);
+
+InstallMethod(ViewString, "for a strong semilattice of semigroups",
+[IsStrongSemilatticeOfSemigroups],
+function(S)
+  local size;
+  size := Size(SemigroupsOfStrongSemilatticeOfSemigroups(S));
+  return Concatenation("<strong semilattice of ",
+                       String(size),
+                       " semigroups>");
+end);
+
+InstallMethod(SSSE,
+"for a strong semilattice of semigroups, a pos int, and an associative elt",
+[IsStrongSemilatticeOfSemigroups, IsPosInt, IsAssociativeElement],
+function(S, n, x)
+  if n > Size(SemigroupsOfStrongSemilatticeOfSemigroups(S)) then
+    ErrorNoReturn("expected second argument to be an integer between 1 and ",
+                  "the size of the semilattice, i.e. ",
+                  Size(SemigroupsOfStrongSemilatticeOfSemigroups(S)));
+  elif not x in SemigroupsOfStrongSemilatticeOfSemigroups(S)[n] then
+    ErrorNoReturn("where S, n and x are the 1st, 2nd and 3rd arguments ",
+                  "respectively, expected x to be an element of ",
+                  "SemigroupsOfStrongSemilatticeOfSemigroups(S)[n]");
+  fi;
+  return Objectify(ElementTypeOfStrongSemilatticeOfSemigroups(S), [S, n, x]);
+end);
+
+InstallMethod(\=, "for SSSEs", IsIdenticalObj,
+[IsSSSERep, IsSSSERep],
+function(x, y)
+  return x![1] = y![1] and x![2] = y![2] and x![3] = y![3];
+end);
+
+InstallMethod(\<, "for SSSEs", IsIdenticalObj, [IsSSSERep, IsSSSERep],
+function(x, y)
+  return (x![2] < y![2]) or (x![2] = y![2] and x![3] < y![3]);
+end);
+
+InstallMethod(\*, "for SSSEs", IsIdenticalObj,
+[IsSSSERep, IsSSSERep],
+function(x, y)
+  local D, meet, maps;
+  D    := SemilatticeOfStrongSemilatticeOfSemigroups(x![1]);
+  meet := PartialOrderDigraphMeetOfVertices(D, x![2], y![2]);
+  maps := HomomorphismsOfStrongSemilatticeOfSemigroups(x![1]);
+  return SSSE(x![1],
+              meet,
+              (x![3] ^ (maps[meet][x![2]])) * (y![3] ^ (maps[meet][y![2]])));
+end);
+
+InstallMethod(ViewString, "for a SSSE",
+[IsSSSERep],
+function(x)
+  return Concatenation("SSSE(", ViewString(x![2]), ", ", ViewString(x![3]), ")");
+end);
+
+InstallMethod(StrongSemilatticeOfSemigroups, "for a SSSE rep",
+[IsSSSERep],
+function(x)
+  return x![1];
+end);
+
+# TODO hash function currently unused
```

Comment:
Does the TODO here mean that you intend to use this?

Reply:
I think we found a way around it, had initially planned to use for equality checking if I remember correctly? Can't think of a use for it now so will remove.

Reply:
Okay, yeah remove this comment and then commented out code, then.